### PR TITLE
Slimming FloatTensor down to a single constructor

### PR DESCRIPTION
**Pull request by iamtrask**

### CODE REVIEW

File: UnityProject/Assets/OpenMined/Syft/Tensor/FloatTensor.cs [hunk 1]
```diff
@@ -76,6 +76,92 @@ public long[] GetIndices (long index)
...
 
+		public FloatTensor (int[] _shape, float[] _data = null, ComputeBuffer _dataBuffer = null, ComputeBuffer _shapebuffer = null, ComputeShader _shader = null, bool _copyData = false)
+{
+			// First: check that shape is valid.
+			if (_shape == null || _shape.Length == 0) {
+				throw new InvalidOperationException ("Tensor shape can't be an empty array.");
+			}	
+
+			// Second: since shape is valid, let's save it
+			shape = (int[])_shape.Clone ();
```
Comment (diegoalejogm) [inline UnityProject/Assets/OpenMined/Syft/Tensor/FloatTensor.cs]: How is _shapebuffer not enough to create a new GPU-memory tensor?

Comment (iamtrask) [reply UnityProject/Assets/OpenMined/Syft/Tensor/FloatTensor.cs]: I was requiring everyone to pass in a shape on the CPU side.

Comment (iamtrask) [reply UnityProject/Assets/OpenMined/Syft/Tensor/FloatTensor.cs]: i suppose it makes sense to allow them to pass either or

Comment (iamtrask) [reply UnityProject/Assets/OpenMined/Syft/Tensor/FloatTensor.cs]: i think i just don't know how much I trust GPUs just yet ;)

**Pull request merged**